### ISSUE TITLE
[Bug]: "Append To Dataset" node lacks error for dataset not found

### ISSUE THREAD

**Issue by ai-made-approachable** [Ironclad/rivet#325]
### What happened?

"Append To Dataset" node keeps running forever, even though it failed  (see screenshot)
This only happens with the node-executor!

### What was the expected functionality?

"Append To Dataset" node should have returned error: "Dataset core_memory not found"

### Describe your environment

MacOS, Node 14.0.0, etc.

### Relevant screenshots

![image](https://github.com/Ironclad/rivet/assets/149181237/e00dc355-9295-426d-9d76-4f7a3274ac53)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by abrenneke** [comment]
Yeah the node executor not finding datasets is... a longstanding problem... definitely shouldn't be an unhandled rejection thuogh.
